sdk/core/azure-core/inc/azure/core/base64.hpp

```diff
@@ -31,23 +34,48 @@ namespace Azure { namespace Core {
 
   public:
     /**
-     * @brief Encodes the vector of binary data into UTF-8 encoded text represented as Base64.
+     * @brief Encodes a vector of binary data using Base64.
      *
-     * @param data The input vector that contains binary data that needs to be encoded.
-     * @return The UTF-8 encoded text in Base64.
+     * @param data The input vector that contains binary data to be encoded.
+     * @return The Base64 encoded contents of the vector.
      */
     static std::string Base64Encode(const std::vector<uint8_t>& data);
 
     /**
-     * @brief Decodes the UTF-8 encoded text represented as Base64 into binary data.
+     * @brief Decodes a Base64 encoded data into a vector of binary data.
      *
-     * @param text The input UTF-8 encoded text in Base64 that needs to be decoded.
+     * @param text Base64 encoded data to be decoded.
      * @return The decoded binary data.
      */
     static std::vector<uint8_t> Base64Decode(const std::string& text);
   };
 
   namespace _internal {
+    /**
+     * @brief Used to convert one form of data  into another, for example encoding binary data into
+     * Base64 encoded octets.
+     *
+     * @note Base64 encoded data is a subset of the ASCII encoding (characters 0-127). As such,
+     * it can be considered a subset of UTF-8.
+     */
+    class Convert final {
+    private:
+      // This type currently only contains static methods and hence disallowing instance creation.
+      /**
+       * @brief An instance of `%Convert` class cannot be created.
+       *
+       */
+      Convert() = default;
+
+    public:
+      /**
+       * @brief Encodes a string using Base64 encoding.
+       *
+       * @param data The input string that contains data to be encoded.
+       * @return The Base64 encoded contents of the vector.
+       */
+      static std::string Base64Encode(const std::string& data);
```

Comment:
Yes, it is needed - the functionality is in core because it's used in core. It's in an `_internal` namespace because it's NOT a part of the public API surface.

The `Azure::Core::_internal::Convert` type adds specializations for base64 and base64url encoding that avoid additional trips through the memory allocator. If we had C++20's `std::span`, I'd be using that, but we don't :(.